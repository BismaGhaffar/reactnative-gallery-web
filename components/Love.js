import React from 'react'
import PropTypes from 'prop-types'
import Heart from 'react-feather/dist/icons/heart'
import { Icon, renderSmallIcon } from './Icon'

const Love = ({ number, onClick }) => (
  <Icon pointer onClick={onClick}>
    {renderSmallIcon(Heart)}
    <Icon.Label>{number}</Icon.Label>
  </Icon>
)

Love.propTypes = {
  number: PropTypes.number,
  onClick: PropTypes.func.isRequired
}

Love.defaultProps = {
  number: 0
}
export default Love
