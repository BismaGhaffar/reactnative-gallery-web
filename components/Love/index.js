import { Heart } from 'react-feather'
import Clickable from '../Clickable'

const Love = props => (
  <Clickable onClick={() => console.log('click')}>
    <Heart {...props} />
  </Clickable>
)

export default Love
