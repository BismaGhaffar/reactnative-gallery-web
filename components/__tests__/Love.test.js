import React from 'react'
import renderer from 'react-test-renderer'
import Love from '../Love'

it('Love can be created', () => {
  const comp = renderer.create(<Love />)
  expect(comp).toBeDefined()
})

it('<Love /> toMatchSnapshot', () => {
  const tree = renderer.create(<Love />).toJSON()
  expect(tree).toMatchSnapshot()
})
