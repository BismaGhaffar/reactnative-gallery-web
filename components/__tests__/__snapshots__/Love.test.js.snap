// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Love /> toMatchSnapshot 1`] = `
<div
  className="Icon-s7tm2sn-0 jglmtW"
  onClick={undefined}
>
  <svg
    fill="none"
    height={22}
    stroke="#fff"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="2"
    viewBox="0 0 24 24"
    width={22}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
    />
  </svg>
  <span
    className="Icon__Label-s7tm2sn-1 grNILv"
  >
    0
  </span>
</div>
`;
